java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/DateConsumer.java

```diff
@@ -85,13 +85,13 @@ public void consume(ResultSet resultSet) throws SQLException {
       Date date = calendar == null ? resultSet.getDate(columnIndexInResultSet) :
           resultSet.getDate(columnIndexInResultSet, calendar);
       if (!resultSet.wasNull()) {
-        int day = (int) TimeUnit.MILLISECONDS.toDays(date.getTime());
-        if (day < 0 || day > MAX_DAY) {
+        long day = TimeUnit.MILLISECONDS.toDays(date.getTime());
+        if (day < Integer.MIN_VALUE || day > MAX_DAY) {
```

Comment:
IMO, the lower bound should not be Integer.MIN_VALUE. Instead, it should be the date corresponding to '-9999-12-31'?

Reply:
Is there context as to why the date range of -9999-12-31 to 9999-12-31? I hadn't seen it in the format spec.

Reply:
It seems different databases have different date ranges. 9999-12-31 is one frequently used (in MySQL & SQL Server). 

https://stackoverflow.com/questions/37138076/get-max-and-min-date-of-java-sql-date

Reply:
Interesting, I'm mostly inclined to just open the bounds to whatever epoch days fit within an integer range given that this is simply a reader and underlying systems have different ranges. Thoughts?

Reply:
Sounds reasonable. However, Integer.MIN_VALUE seems overly conservative, as it forces us to use big int comparisons.

Reply:
Yeah, but otherwise we can end up with particularly bad data. I've changed the condition to just be something similar to Math.toExactInt where we see if the casted value is the same.

Without doing the big int comparison I think we run the risk of wrapping around into a totally valid positive range (e.g. 2*Integer.MAX_VALUE + x)